net/net_tcp.c

```diff
@@ -461,10 +462,14 @@ int tcp_conn_get(unsigned int id, struct ip_addr* ip, int port,
 					a, a->parent, a->parent->id, a->port,
 					a->parent->rcv.src_port);
 				print_ip("ip=",&a->parent->rcv.src_ip,"\n");
+				if (send_sock != NULL && a->parent->send_sock != NULL)
+					print_ip("requered send_sock ip=",&send_sock->address,"\n");
+					print_ip("found send_sock ip=",&a->parent->send_sock->address,"\n");
 #endif
 				c = a->parent;
 				if (c->state != S_CONN_BAD &&
 				    c->flags&F_CONN_INIT &&
+				    send_sock == a->parent->send_sock &&
```

Comment:
If a socket is not forced (thus `send_sock` is NULL), then we might want to use any connection towards that destination, don't we?

Reply:
Fixed